### PR TITLE
[Bugfix:System] Disabled audio, usb, uart on vagrant vb

### PR DESCRIPTION
**Pull request by Yujjio**
### Please check if the PR fulfills these requirements:
* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant

### What is the current behavior?
Nothing inside the `vagrant.configure`

Windows users may timeout at the `SSH auth method: private key` part when installing Submitty VM.

Related links:
* https://github.com/hashicorp/vagrant/issues/11777
* https://bugs.launchpad.net/cloud-images/+bug/1829625
* https://bugs.launchpad.net/cloud-images/+bug/1874453

### What is the new behavior?
Disabled unnecessary VM functions, including audio, USB, and UART1-4.

### Other information?
Tested in 2 Windows laptops and no timeout happened

### CODE REVIEW

**Comment by bmcutler** [comment]
@Yujjio Please read
https://submitty.org/developer/getting_started/how_to_contribute

I believe this PR would be better titled [Bugfix:System]
(DevDependency is for 3rd party library updates)

**Comment by Yujjio** [comment]
> @Yujjio Please read https://submitty.org/developer/getting_started/how_to_contribute
> 
> I believe this PR would be better titled [Bugfix:System] (DevDependency is for 3rd party library updates)

Got it! Changed the PR title

File: Vagrantfile [hunk 1]
```diff
@@ -77,6 +77,17 @@ def mount_folders(config, mount_options)
 end
 
 Vagrant.configure(2) do |config|
+  config.vm.boot_timeout=600
+  config.vm.provider :virtualbox do |v|
+    v.customize ["modifyvm", :id, "--uart1", "0x3F8", "4"]
+    v.customize ["modifyvm", :id, "--uartmode1", "file", File::NULL]
+    v.customize ["modifyvm", :id, "--audio", "none"]
```
Comment (d0ngeun) [inline Vagrantfile:84]: From my tests, these modifications are not really necessary. The first two lines with uart1 and uartmode1 seem to fix the freeze on their own.

**Review by d0ngeun — COMMENTED** [review]
The first 2 modifications with uart1 and uartmode1 are the only necessary ones needed to fix the freeze bug. I suggest removing the other vm customizations.

File: Vagrantfile [hunk 2]
```diff
@@ -77,6 +77,17 @@ def mount_folders(config, mount_options)
 end
 
 Vagrant.configure(2) do |config|
+  config.vm.boot_timeout=600
```
Comment (MasterOdin) [inline Vagrantfile:80]: This is already done on line 94 and can be removed.

File: Vagrantfile [hunk 3]
```diff
@@ -77,6 +77,17 @@ def mount_folders(config, mount_options)
...
+  config.vm.provider :virtualbox do |v|
```
Comment (MasterOdin) [inline Vagrantfile:81]: We already have a `config.vm.provider` block and I'd suggest merging the two to have just one block.

Comment (Yujjio) [reply Vagrantfile:81]: Yes I just saw that and I will merge it. Thanks

**Comment by Yujjio** [comment]
> The first 2 modifications with uart1 and uartmode1 are the only necessary ones needed to fix the freeze bug. I suggest removing the other vm customizations.

I tried to only add the first 2 lines at first but it still will timeout from time to time, by adding the other modification, I never meet timeout again so I think these lines are also helpful

File: Vagrantfile (hunk 2 above)
Comment (Yujjio) [reply Vagrantfile:80]: which version's line 94? I didn't see that in the main branch or in my branch

Comment (MasterOdin) [reply Vagrantfile:80]: Sorry, meant line 84:

https://github.com/Submitty/Submitty/blob/4a6309822740858cab5d1052768fd33815ea0aec/Vagrantfile#L84

Comment (Yujjio) [reply Vagrantfile:80]: OK, I see. Will modify it soon. Thanks

**Review by d0ngeun — APPROVED** [review]

**Review by MasterOdin — APPROVED** [review]

**Comment by MasterOdin** [comment]
Thanks for the PR to improve the situation on vagrant!

**Pull request merged**